aios/trajectory.py

```diff
@@ -0,0 +1,186 @@
+            self.v + deltaV <= self.MAX_LAUNCH_SPEED
+        )
+        opti.subject_to(
+            0 < self.theta + deltaTheta 
+        )
+        
+        opti.subject_to(
+            self.theta + deltaTheta < math.pi / 2
+        )
+
+        opti.subject_to(
+            -1 * math.pi / 2 < self.phi + deltaPhi 
+        )
+
+        opti.subject_to(
+            self.phi + deltaPhi < math.pi / 2
+        )
+
+        opti.subject_to(
+            self.MAX_HEIGHT - (self.NOTE_THICKNESS / 2) >
+           ( (self.v + deltaV) * casadi.sin(deltaTheta + self.theta) * (self.y - self.OPENING_GAP)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy ) )
+           - 0.5 * self.GRAVITATIONAL_ACCELERATION * ((self.y - self.OPENING_GAP)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )) ** 2
+        )
+
+        opti.subject_to(
+            self.MIN_HEIGHT + (self.NOTE_THICKNESS / 2) >
+           ( (self.v + deltaV) * casadi.sin(deltaTheta + self.theta) * (self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy ) )
+           - 0.5 * self.GRAVITATIONAL_ACCELERATION * ((self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )) ** 2
+        )
+
+        opti.subject_to(
+            0 <=
+           ( (self.v + deltaV) * casadi.sin(deltaTheta + self.theta) * (1)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy ) )
+           - self.GRAVITATIONAL_ACCELERATION * (self.y)/ ( ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )) ** 2
+        )
+
+        opti.subject_to(
+            (-1 * self.SPEAKER_WIDTH + self.NOTE_OUTER_RADIUS) / 2 < self.x + (( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.sin(self.phi + deltaPhi)) + self.vx) * (self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )
+        )
+
+        opti.subject_to(
+            (self.SPEAKER_WIDTH - self.NOTE_OUTER_RADIUS) / 2 > self.x + (( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.sin(self.phi + deltaPhi)) + self.vx) * (self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )
+        )
+
+
+        
+        opti.solver('ipopt')
+        try:
+            solutions = opti.solve()
+        except:
+            return ValueError("No calculations for shooting available at this position.")
+
+
+        return {
+            'speed' : round(solutions.value(deltaV) + self.v, 3), 
+            'angle' : round(solutions.value(deltaTheta) + self.theta, 3),
+            'heading' : round(solutions.value(deltaPhi) + self.phi , 3)
+        }
+
+    def canShootWhileStationary(self) -> bool:
```

Comment:
this is really just a function of an x and y coordinate, as far as i can tell, and doesn't require any of the other variables of a trajectory. Could you make it a general method of the class (so in java, it would be static) and have it take in an x and y coordinate instead of a trajectory?